### PR TITLE
Addition of IDEs and Editors

### PR DESCRIPTION
**Pull request by lourieder**
I added a list with IDEs and Text Editors. @gtema requested changes on a previos PR were considered

### CODE REVIEW

**Review by gtema — CHANGES_REQUESTED** [review]
Can you please ensure build is passing?

**Comment by lourieder** [comment]
I now can esure the build is indeed **passing**. 
There were some minor issues I fixed

File: doc/source/code_editors.rst [hunk 1]
```diff
@@ -1,3 +1,67 @@
 ============
 Code Editors
 ============
+
+==============================
```
Comment (gtema) [inline doc/source/code_editors.rst:5]: This is wrong: you make 2 titles of the doc (look at https://thomas-cokelaer.info/tutorials/sphinx/rest_syntax.html#id8 how this work). In principle you only drop upper "=" above `IDE and ...` and that's it

**Review by gtema — CHANGES_REQUESTED** [review]

File: doc/source/code_editors.rst (hunk 1 above)
Comment (lourieder) [reply doc/source/code_editors.rst]: Thanks for the input I changed it accordingly

**Review by gtema — APPROVED** [review]
Please in the follow-up either split it into IDEs and Editors or drop chapter "IDE and Editor recommendations" at all - this doesn't really look right now

**Pull request merged**